lib/app/sequence_base.rb

```diff
@@ -367,119 +362,69 @@ def self.extends_sequence(klass)
       def self.test(name, &block)
         @@test_index += 1
 
-        test_index = @@test_index
-
-        test_method = "#{@@test_index.to_s.rjust(4, '0')} #{name} test".downcase.tr(' ', '_').to_sym
-        @@test_metadata[sequence_name] << { name: name,
-                                            test_index: test_index,
-                                            required: true,
-                                            versions: FHIR::VERSIONS }
-
-        current_test = @@test_metadata[sequence_name].last
-
-        wrapped = lambda do
-          instance_eval(&block) if @metadata_only # just run the test to hit the metadata block
+        tests << InfernoTest.new(name, @@test_index, &block)
+      end
 
+      def wrap_test(test)
+        lambda do |sequence|
```

Comment:
Any ideas where this lambda's argument is coming from?

Reply:
What do you mean?  You don't use that argument anywhere inside the lambda, and you aren't passing it into the lambda when calling it.  Do you want to use that argument but can't figure out how to pass it in?  (I don't see why you don't just remove the argument)

Reply:
What you are saying is exactly how I understand it, but lambdas enforce the number of arguments and if I remove the argument, I get an `ArgumentError`. I could probably just replace the lambda with a Proc, but I was hoping to be able to remove the argument from wherever it's coming from instead.

Reply:
See https://stackoverflow.com/questions/15304485/why-does-instance-eval-succeed-with-a-proc-but-not-with-a-lambda

`instance_eval` passes the instance of the object as the first parameter.  This seems to get in some gnarly aspects of binding blocks of code to objects.  Accepting `_sequence` seems fine by me.  Or you can change to `Proc` probably.  Or you can try `instance_exec`.  But there might be side effects that prevent those other options working the way we want (so I think its just best to leave as-is because it works).

Reply:
Thank you! `instance_exec` solved the problem, and I think it is the cleanest solution.